### PR TITLE
Fix for: mWeb/Chrome - Payment - The keyboard overlaps the "Make default payment method" button

### PR DESCRIPTION
**Pull request by Ollyws**
### Fixed Issues

$ https://github.com/Expensify/App/issues/14003   
PROPOSAL: https://github.com/Expensify/App/issues/14003#issuecomment-1410056441

### Tests

1. Go to Settings -> Payment
2. Tap on the bank account that doesn't have the "default" label
3. Tap on "Make default payment method" button
4. Ensure the keyboard does not overlap the "Make default payment method" button.
- [x] Verify that no errors appear in the JS console

### Offline tests

1. Go to Settings -> Payment
2. Tap on the bank account that doesn't have the "default" label
3. Tap on "Make default payment method" button
4. Ensure the keyboard does not overlap the "Make default payment method" button.
### QA Steps

1. Go to Settings -> Payment
2. Tap on the bank account that doesn't have the "default" label
3. Tap on "Make default payment method" button
4. Ensure the keyboard does not overlap the "Make default payment method" button.
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/11609254/215994253-6c7274f2-3da5-48bd-833c-b2bca6b3b891.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/11609254/215994282-9db2a399-f1f4-49c4-9160-8df45ad9b989.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/11609254/215994318-9c35d4f7-5abe-4fe5-bac3-91519d79a143.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/11609254/215994369-34feea8e-7937-4f22-af12-0f9fbe39bc96.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/11609254/215994428-783c2230-aa5a-46cb-9689-da180f0a02a5.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/11609254/215994456-9a0f39fa-c1f3-47c2-b73e-7634b44d04fb.mp4

</details>

### CODE REVIEW

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

File: src/components/KeyboardSpacer/index.android.js
Comment (jasperhuangg) [inline src/components/KeyboardSpacer/index.android.js:21]: Why are we deleting this file? This fix is for mWeb chrome, and shouldn't affect behavior on Android native (AFAIK)

File: src/components/withViewportOffsetTop.js
Comment (jasperhuangg) [inline src/components/withViewportOffsetTop.js:8]: Let's add a comment here explaining what this prop is, that way if anyone else wants to use this component it's very clear to them

**Review by jasperhuangg — CHANGES_REQUESTED** [review]
Mostly looks good! Just a few minor comments, and please address @aimane-chnaif's feedback

File: src/components/KeyboardSpacer/index.android.js
Comment (jasperhuangg) [reply src/components/KeyboardSpacer/index.android.js:21]: understood! all good, thanks for linking everything and clarifying

**Comment by jasperhuangg** [comment]
@aimane-chnaif @Ollyws I agree that should try diff 2

**Review by aimane-chnaif — DISMISSED** [review]
Latest code changes look good to me and tests work as expected, except [3 issues](https://github.com/Expensify/App/pull/14725#issuecomment-1415664589) I mentioned above.

**Comment by jasperhuangg** [comment]
> Issues to note (out of scope):
> 
> * [iOS -  App freezes when we close default payment when touch outside of modal - reported by @dhairyasenjaliya #10509](https://github.com/Expensify/App/issues/10509)
> * android chrome - bottom padding above keyboard while scrolling (this is new issue caused by this PR, not happening on staging)
> 
>  mchrome.mp4 
> * iOS safari - weird scroll behavior - button keeps moving, bottom padding above keyboard while scrolling (this is happening on staging)
> 
>  scroll.mp4 
>  ios.mp4

@Ollyws is there really no way to fix these issues before merging this PR? Like disabling scroll while the modal is open?

**Comment by jasperhuangg** [comment]
@Ollyws Looks like this is already a known issue and could potentially be fixed by solving that other issue later on. That makes sense.

Thanks @aimane-chnaif for clarifying. Let's also update https://github.com/Expensify/App/issues/14718 to include the fact that this issue will crop up in the make default payment method modal once this PR is merged/deployed.

File: src/components/Modal/BaseModal.js
Comment (jasperhuangg) [inline src/components/Modal/BaseModal.js:80]: I don't really like the name `extraModalStyles`, can we come up with something a bit more descriptive?

**Review by jasperhuangg — CHANGES_REQUESTED** [review]

**Comment by jasperhuangg** [comment]
@aimane-chnaif will do once this is deployed!

File: src/components/Modal/BaseModal.js
Comment (jasperhuangg) [reply src/components/Modal/BaseModal.js:80]: Thanks for your patience guys have been out sick.

I like `outerStyle`? And I think we can rename `containerStyle` to `innerContainerStyle` since the container those styles are being applied to lives inside the modal.

Comment (jasperhuangg) [reply src/components/Modal/BaseModal.js:80]: @Ollyws yeah go ahead! Sorry for the delay

**Comment by jasperhuangg** [comment]
Just replied to @Ollyws's comment [here](https://github.com/Expensify/App/pull/14725#discussion_r1098992146), I'll be back to approve and merge once those changes are made, thanks for your patience guys

**Review by jasperhuangg — APPROVED** [review]

**Pull request merged**
